app/api/users.py

```diff
@@ -108,9 +108,12 @@ class UserDetail(ResourceDetail):
     """
 
     def before_get(self, args, kwargs):
-        if current_user.is_admin or current_user.is_super_admin or current_user:
+        if current_user:
             self.schema = UserSchema
-        else:
+        elif 'Authorization' in request.headers and (has_access('is_admin') or has_access('is_super_admin')):
+            self.schema = UserSchema
+        elif 'Authorization' in request.headers and (kwargs['event_id'] and has_access('is_owner',
+                                                     event_id=int(kwargs['event_id']))):
```

Comment:
We want it to be visible to all users, not just owner of the event

Reply:
Yess It is visible to all users, I've added a screenshot above. This check is for if the user itself request for owner then we can provide the complete info because we're excluding this case from the decorator we need to check for is_user_itself somewhere, this is the place

Reply:
Then `or` should be used. It's looking like 2 different branches

Reply:
Implementing this